test/test_examples.py

```diff
@@ -42,6 +43,47 @@ def test_handle_directory_with_log_file(self):
             [gr.media_data.BASE64_IMAGE, "hi"],
         ]
 
+    @pytest.mark.asyncio
+    async def test_no_preprocessing(self):
+        with gr.Blocks():
+            image = gr.Image()
+            textbox = gr.Textbox()
+
+            examples = gr.Examples(
+                examples=["test/test_files/bus.png"],
+                inputs=image,
+                outputs=textbox,
+                fn=lambda x: x,
+                cache_examples=True,
+                preprocess=False,
+            )
+
+        await examples.cache_interface_examples()
+        prediction = await examples.load_from_cache(0)
+        assert isinstance(prediction[0], str)
```

Comment:
I think I get the same prediction no matter if preprocess is true or false

Reply:
I think there are two problems:
1. We call preprocessing prior to `predict_example` and store it in `self.processed_examples`
2. We're reusing the cache between runs of this test. I would mock out `CACHE_DIR` to make sure all runs get a fresh cache.

Reply:
Interesting let me take a look. To clarify, you mean that we call _post_-processing prior to `predict_example`, right? That's designed to convert from a human-readable version to a base64 so that the sample is ready to be _pre_-processed.

Reply:
Ok fixed now. The examples were being cached twice so removed that extra line and mocked out the cache folder. Now I can confirm that setting `preprocess=True` causes the test to fail.